x-pack/legacy/plugins/index_management/public/components/mappings_editor/reducer.ts

```diff
@@ -240,6 +255,20 @@ export const reducer = (state: State, action: Action): State => {
         },
       };
     }
+    case 'fieldsJsonEditor.update':
+      return {
```

Comment:
Instead of making 2 copies of the state, can we create like in other places an intermediary `nextState` and provide it to the `determineIfValid ()` ?